src/Streamly/Internal/FileSystem/Handle.hs

```diff
@@ -257,9 +258,34 @@ readChunksWithBufferOf = Unfold step return
 --
 --
 {-# INLINE_NORMAL readChunksFromToWith #-}
-readChunksFromToWith :: -- MonadIO m =>
+readChunksFromToWith :: MonadIO m =>
     Unfold m (Int, Int, Int, Handle) (Array Word8)
-readChunksFromToWith = undefined
+readChunksFromToWith = Unfold step inject
+
+    where
+
+    inject (from, to, buffSize, h) = do
+        liftIO $ hSeek h AbsoluteSeek $ fromIntegral from
+        return (to - from + 1, buffSize, h)
+
+    {-# INLINE_LATE step #-}
+    step (remaining, buffSize, h) =
+        if remaining <= 0
+        then return D.Stop -- if range is unsafe return empty stream
+        else
+            do
+            arr <- getChunk buffSize h
+            return $
+                case A.length arr of
+                    0 -> D.Stop
+                    len ->
+                        if (remaining - len) >= 0
+                        then
+                            D.Yield arr (remaining - len, buffSize, h)
+                        else
+                            D.Yield
+                            (A.unsafeSlice 0 remaining arr)
+                            (0, 0, h)
```

Comment:
If we use `min` in `getChunk` then this should not occur. We can put a `error "Bug: ...` message here.